CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+cmake_minimum_required(VERSION 3.0)
+
+project(OpenCLHeaders
+  VERSION 2.2
+  LANGUAGES C # Ideally should be NONE, but GNUInstallDirs detects platform arch using try_compile
+  # https://stackoverflow.com/questions/43379311/why-does-project-affect-cmakes-opinion-on-cmake-sizeof-void-p
+)
+
+add_library(Headers INTERFACE)
+target_include_directories(Headers
+  SYSTEM
+  BEFORE
+  INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+)
+
+install(
+  TARGETS Headers
+  EXPORT OpenCLHeadersTargets
+)
+
+include(GNUInstallDirs)
+
+install(
+  DIRECTORY CL
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+export(
+  EXPORT OpenCLHeadersTargets
+  FILE ${PROJECT_BINARY_DIR}/OpenCLHeaders/OpenCLHeadersTargets.cmake
+  NAMESPACE OpenCL::
+)
+file(
+  WRITE ${PROJECT_BINARY_DIR}/OpenCLHeaders/OpenCLHeadersConfig.cmake
+  "include(\"\${CMAKE_CURRENT_LIST_DIR}/OpenCLHeadersTargets.cmake\")"
+)
+
+set(config_package_location ${CMAKE_INSTALL_DATADIR}/cmake/OpenCLHeaders)
+install(
+  EXPORT OpenCLHeadersTargets
+  FILE OpenCLHeadersTargets.cmake
+  NAMESPACE OpenCL::
+  DESTINATION ${config_package_location}
+)
+install(
+  FILES ${CMAKE_CURRENT_BINARY_DIR}/OpenCLHeaders/OpenCLHeadersConfig.cmake
+  DESTINATION ${config_package_location}
+)
+
+unset(CMAKE_SIZEOF_VOID_P)
```

Comment:
Why?

Reply:
I took this from the the previous contributor @sl1pnk07, I believe this is in order to have packages always install to `lib` and not `lib64` on 64-bit platforms. I'm not entirely sure. If this is the reason, it would cause problem only if people installed their own builds of something into system paths. That is a slippery slope and generally calling for trouble.

Reply:
I think we should leave a TODO or create an issue if we don't really understand this fully.